internal/app/app_darwin.go

```diff
@@ -0,0 +1,189 @@
+package app
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/ActiveState/cli/internal/assets"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/installation"
+	"github.com/ActiveState/cli/internal/strutils"
+)
+
+const (
+	execFileSource   = "exec.sh.tpl"
+	launchFileSource = "com.activestate.platform.app.plist.tpl"
+	iconFile         = "icon.icns"
+	assetAppDir      = "placeholder.app"
+)
+
+func (a *App) install() error {
+	// Create all of the necessary directories and files in a temporary directory
+	// Then move the temporary directory to the final location which for macOS will be the Applications directory
+	tmpDir, err := ioutil.TempDir("", fmt.Sprintf("%s-", a.Name))
+	if err != nil {
+		return errs.Wrap(err, "Could not create temporary directory")
+	}
+	defer os.RemoveAll(tmpDir)
+
+	tmpAppPath := filepath.Join(tmpDir, fmt.Sprintf("%s.app", a.Name))
+	if err := fileutils.Mkdir(tmpAppPath); err != nil {
+		return errs.Wrap(err, "Could not create .app directory")
+	}
+
+	appDir, err := assets.OpenFile(assetAppDir)
+	if err != nil {
+		return errs.Wrap(err, "Could not read app directory asset")
+	}
+
+	info, err := appDir.Stat()
+	if err != nil {
+		return errs.Wrap(err, "Could not stat app directory asset")
+	}
+
+	err = generateAppDir(tmpAppPath, info.Name())
+	if err != nil {
+		return errs.Wrap(err, "Could not generate app directory")
+	}
```

Comment:
This didn't end up where I would have liked it. Unfortunately, the embedded file system doesn't have enough functionality to simply copy one directory to another. So I can reflect the directory structure in our assets are recreate it from that.

Reply:
It doesn't have a convenient function for it, but you should be able to do it by iterating over `fs.ReadDir()` and then copying the individual files it comes across and entering into subdirs. You can probably repurpose this function to allow supplying the embed fs:

https://github.com/ActiveState/cli/blob/34b6f33e804560166c6df72bf97ddb24f3ddf9b2/internal/fileutils/fileutils.go#L690-L690